packages/hardhat/contracts/KanbanFactory.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0 <=0.8.0;
+
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract KanbanFactory {
+
+  event kanbanCreated(uint id, address creator, address instance);
+
+  using Counters for Counters.Counter;
+
+  Counters.Counter private _kanbanIds;
+  mapping(uint => address) public kanbanInstances;
+  address private baseKanbanAddress;
+  
+  constructor(address _baseKanbanAddress) {
+    baseKanbanAddress = _baseKanbanAddress;
+  }
+
+  function createKanban() public {
+    address instance = Clones.clone(baseKanbanAddress);
+    kanbanInstances[_kanbanIds.current()] = instance;
+    emit kanbanCreated(_kanbanIds.current(), msg.sender, instance);
```

Comment:
(Sorry if this is totally wrong) 
I think this introduces a re-entrancy vulnerability. E.g. if the contract is called multiple times quickly we'll create multiple kanbans and lose some of them. 

In general programming I would have said that having a predictable increasing ID is a bad thing too, but I guess that whatever ID we pick, it's all going to be visible on the chain anyway :) 

But maybe using a securely generated random ID here would avoid collisions.

Reply:
"But maybe using a securely generated random ID here would avoid collisions." - 
I think we can simply return or emit the address where its deployed, if the id creates a collision